tests/zfs-tests/include/blkdev.shlib

```diff
@@ -80,7 +80,23 @@ function block_device_wait
 		[[ $elapsed > 60 ]] && \
 		    log_note udevadm settle time too long: $elapsed
 	elif is_freebsd; then
-		sleep 3
+		if [[ ${#@} -eq 0 ]]; then
```

Comment:
This seems generally useful. Should it be unconditional?

Reply:
Can you clarify? The `sleep 3` is not so useful, rather it's more a burden I'm trying to avoid where possible with this change.

Reply:
Sure, I wasn't referring specifically to the sleep, but rather the polling that follows. I've seen rare ZoL failures where it appears the device file is not yet available despite having returned from block_device_wait.

If you'd rather keep this as is, that's fine; I can look at making this change later on.

Reply:
Ah I can do that.